App/Converters/DependencyObjectSelector.cs

```diff
@@ -0,0 +1,155 @@
+using System;
+using System.Linq;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml.Markup;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Media;
+
+namespace Coder.Desktop.App.Converters;
+
+// This file uses manual DependencyProperty properties rather than
+// DependencyPropertyGenerator since it doesn't seem to work properly with
+// generics.
+
+public class DependencyObjectSelectorItem<TK, TV> : DependencyObject
```

Comment:
What is this used for?

Reply:
I can add better comments, but...

`DependencyObjectSelector` contains multiple `DependencyObjectSelectorItem`s, each with a key and a value. The `DependencyObjectSelector` has a dependency property on it called `SelectedKey`, which when updated from the view will cause `SelectedObject` to be updated to the value of the matching `DependencyObjectSelectorItem`

Reply:
Ah! If I'm understanding you correctly, this is a map, but for DependencyProperties so it does the right stuff to wire up to the presentation framework. And, the reason a Converter won't work is that the converter approach only triggers updates when the "key" side gets updated, whereas for themed colors, the "values" could also update on a theme change, and we want to propagate that change thru.

Reply:
This class could really use a unit test that updates both keys and values and asserts that the changes propagate.

Reply:
Discussed on Slack, due to issues it's difficult to test this currently: #62